### PR TITLE
mgmt, update test on MongoCluster

### PR DESCRIPTION
**Pull request by v-hongli1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/mongocluster/tests.mgmt.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+extends:
+  template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
+  parameters:
+    ServiceDirectory: mongocluster
+    Artifacts:
+      - name: azure-resourcemanager-mongocluster
+        groupId: com.azure.resourcemanager
+        safeName: azureresourcemanagermongocluster
+    Clouds: 'Public'
+    # Only run tests on Windows to save cost.
```
Comment (XiaofeiCao) [inline sdk/mongocluster/tests.mgmt.yml:14]: @weidongxu-microsoft Do we need `useFederatedAuth: true`?

Comment (weidongxu-microsoft) [reply sdk/mongocluster/tests.mgmt.yml:14]: I think it is "yes".

File: sdk/mongocluster/test-resources.bicep [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  name: guid('contributorRoleId${resourceGroup().name}')
+  properties: {
+    roleDefinitionId: contributorRoleId
+    principalId: testApplicationOid
+  }
+}
+
+output AZURE_TENANT_ID string = tenantId
+output AZURE_CLIENT_ID string = testApplicationId
+output AZURE_CLIENT_SECRET string = testApplicationSecret
```
Comment (weidongxu-microsoft) [inline sdk/mongocluster/test-resources.bicep:25]: Do not output this 2. No secret anymore.

**Review by weidongxu-microsoft — CHANGES_REQUESTED** [review]
No client secret allowed.

File: sdk/mongocluster/test-resources.bicep (hunk 2 above)
Comment (v-hongli1) [reply sdk/mongocluster/test-resources.bicep]: Removed it in the new version.

File: sdk/mongocluster/tests.mgmt.yml (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/mongocluster/tests.mgmt.yml:14]: @v-hongli1 Please use the latest template in your previous PR:
https://github.com/Azure/azure-sdk-for-java/pull/40942/files

**Comment by XiaofeiCao** [comment]
I'm not seeing mgmt test pipeline being created.

**Comment by v-hongli1** [comment]
> I'm not seeing mgmt test pipeline being created.

I also find it a bit strange. Analyzing from the execution log of `prepare-pipelines`, the code used when reconfiguring the pipeline is the `main` branch of `Azure/azure-sdk-for-java`, and the changes in this PR have not been merged. 
The log of `prepare-pipelines` as follow as:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4146613&view=results

**Comment by XiaofeiCao** [comment]
I'm not sure, But I believe we should fix it before merging the PR. Maybe there's a bug in the script.
Let's wait for Weidong's comments first.

**Comment by v-hongli1** [comment]
@weidongxu-microsoft  @XiaofeiCao 
The mgmt test pipeline has been created by `prepare-pipelines` , but I cannot confirm whether the live test case added in the PR has been executed. Can you help me confirm this ?

**Comment by weidongxu-microsoft** [comment]
Looks good on tests column
image

Though I also didn't see any log output in pipeline.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**